Scripts/LoadWald.R

```diff
@@ -1,50 +1,78 @@
     joint.wald.test        = numeric(6)
-    names(joint.wald.test) = c("Name","W","p-value", "df", "H0" , "Decision")
+    names(joint.wald.test) = c("Test","W","p-value", "df", "H0" , "Decision")
     beta                   = model.summary$coefficients[,1]
     Var_beta_est           = vcov(model.summary)
     
+    # Set up significance level
+    signf.level  = if (is.null(signf.level)){
+        signf.level = 0.95 # 95% significance level as default
+    } else {
+        signf.level = signf.level}
+    
+    # Set up test restrictions
+    spec   = if (is.null(spec)){
+        spec = 1: length(beta) # default joint is significance test
+    } else {
+        spec = spec}
+    
+    
     # Wald test statistic
     W = t(beta[spec]) %*% solve(Var_beta_est[spec,spec]) %*% beta[spec]
     
     # Set up test output
-    chi2               = qchisq(signf.l, df=length(spec))
+    chi2               = qchisq(signf.level, df = length(spec))
     pval               = 1-pchisq(W,length(spec))
-    joint.wald.test[1] = "Chi2 test"
+    joint.wald.test[1] = "Wald"
     joint.wald.test[2] = format(   W, digits = 4) 
     joint.wald.test[3] = format(pval, digits = 4)
     joint.wald.test[4] = length(spec)
     joint.wald.test[5] = "b equal to 0"
-    joint.wald.test[6] = ifelse(pval <= 1- signf.l, "Reject H0", "Cannot reject H0")
+    joint.wald.test[6] = ifelse(pval <= 1- signf.level, "Reject H0", "Cannot reject H0")
     joint.wald.test
 }
 
 ################################################################################
 ## Wald Test for linear hypothesis
 
 # Note: This a general Wald test for linear hypothesis of model beta coefficients
-# m = number of hypothesis
-# beta is vector of coefficients
-# R is Jacobian matrix of size m x n_beta, r is restriction function size m x 1
-# Example: H0: b1 + b2 = 1 and b1 = 0  (2 resitrciotns, m = 2)
+# m = number of restrictions
+# Beta is vector of coefficients of size k x 1 
+# Signf.level is the desired significance level (between 0 and 1). Default: 0.95
+# R is Jacobian matrix of size m x k 
+# r is restriction function size m x 1
+# Example: H0: b1 + b2 = 1 and 2*b1 = 0  (2 restrictions, m = 2)
 # R = 1 1 0 0 ...0      and r = 1
-#     1 0 0 0 ...0              0 
+#     2 0 0 0 ...0              0 
 #ToDo: incorporate error/warnings if R and r are not specified correctly (i.e. wrong dimensions)
 
 
-general.wald.test = function(model.summary, signf.l, R = NULL, r = NULL){
+general.wald.test = function(model.summary, signf.level = NULL, R = NULL, r = NULL){
     
     # Define test elements
     general.wald.test        = numeric(6)
-    names(general.wald.test) = c("Name","W","p-value", "df", "H0" , "Decision")
+    names(general.wald.test) = c("Test","W","p-value", "df", "H0" , "Decision")
     beta                     = model.summary$coefficients[, 1]
     Var_beta_est             = vcov(model.summary)
     
+    # Set up significance level
+    signf.level  = if (is.null(signf.level)){
```

Comment:
You actually don't need to add these conditional statements. In R you can actually specify default function arguments in the function call.

So in the function call just write:
`function(model.summary, signf.level = 0.95, ...)`

Doing so will set `signf.level` to 0.95 if it isn't explicitly set, otherwise it will be what the user sets in the function argument.